features/gui/calculator/technology.lua

```diff
@@ -0,0 +1,456 @@
+                breakdown.add { type = 'label', caption = avg.relative }
+                breakdown.add { type = 'label', caption = avg.absolute }
+            end
+
+            breakdown.visible = (technology_name ~= nil)
+        end
+
+        do --- Technology research path
+            local label = body.add {
+                type = 'label',
+                style = 'bold_label',
+                caption = toggled[player.index] and on_technologies or off_technologies,
+                name = toggle_list_button_name,
+                tooltip = 'Hide/Show technologies list',
+            }
+            data.label = label
+            Gui.set_data(label, data)
+
+            local deep = body.add { type = 'frame', style = 'deep_frame_in_shallow_frame_for_description', direction = 'vertical' }
+            Gui.set_style(deep, { padding = 0, minimal_height = 4 })
+
+            local list = deep.add { type = 'scroll-pane', vertical_scroll_policy = 'never' }.add { type = 'table', style = 'table_with_selection', column_count = 3 }
+            Gui.set_style(list.parent, { horizontally_squashable = false })
+
+            list.visible = toggled[player.index] or false
+            data.list = list
+            Gui.set_data(list, data)
+
+            for _, t in pairs(info.research_path) do
+                shortcut_button(list, t)
+                list.add { type = 'label', caption = t.localised_name }
+                show_ingredients(list, t.ingredients)
+            end
+        end
+    end
+end
+
+Gui.on_click(close_button_name, function(event)
+    Gui.destroy(Gui.get_data(event.element).frame)
+end)
+
+Gui.on_custom_close(main_frame_name, function(event)
+    Gui.destroy(event.element)
+end)
+
+Gui.on_click(history_back_button_name, function(event)
+    local h = get_history(event.player_index)
+    h:previous()
+    draw(event.player)
+end)
+
+Gui.on_click(history_forward_button_name, function(event)
+    local h = get_history(event.player_index)
+    h:next()
+    draw(event.player)
+end)
+
+Gui.on_elem_changed(select_button_name, function(event)
+    set_history(event.player_index, event.element.elem_value)
+    draw(event.player)
+end)
+
+Gui.on_click(toggle_list_button_name, function(event)
+    local data = Gui.get_data(event.element)
+    data.list.visible = not data.list.visible
+    toggled[event.player_index] = data.list.visible
+    data.label.caption = data.list.visible and on_technologies or off_technologies
+end)
+
+Gui.on_click(shortcut_button_name, function(event)
+    if get_history(event.player_index):get() == event.element.tags.name then
+        return
+    end
+    set_history(event.player_index, event.element.tags.name)
+    draw(event.player)
+end)
+
+-- == COMMANDS ================================================================
+
+Command.add('calculator-technology', {
+    description = 'Computes the cost in science packs to research target technology',
+    arguments = { 'technology' },
```

Comment:
Maybe this argument should be optional.
```suggestion
    arguments = { 'technology' },
    default_values = { technology = '' },
```

Reply:
Ah yes, ty. I was wondering what the correct syntax was for optional params in commands definitions, I completely missed in the module's docs